### ISSUE TITLE
Bugs and some questions about the paper

### ISSUE THREAD

**Issue by wenbo524822927** [Hojat-Asgariandehkordi/Plane-wave-denoising-diffusion#2]
1. In your dataset.py code, there are three output of the __getitem__(), so on the train.py, batch[3] is out of range. Based on the paper and the function get_loss and forward_diffusion_sample, I guess the batch[3] should be the difference between input1 and input2, which is X1-X0 in the equation(19). 
2. t = torch.randint(1, SIGMA_R + 1, (BATCH_SIZE,)).to(device).view(BATCH_SIZE, 1, 1, 1), if the SIGMA_R=1, so the t = 1. I do not know what unet.py you use, but the t can not always be 1, if you train a DDPM-like model. it is so confusing if set both SIGMA_R and sigma_ratio to 10, in this way, we get a correct V but the t is not a float.

3. In the paper, "Fig. 5 depicts the selected mask and its resulting simulated 75 compounding PW image for two samples in the training set. The compounding angle range started from -16 to 16 degree with the difference of 0.5 degree for each PW." if the simulated angle is from -16 to 16 and step 0.5, there are total 65 angles, not 75.

4. In the def _process_image(self, image) funciton, I think you try to add Hibert transfer,  envelope detection and dB scale. After the process, the image of PW and CPW is not a Gaussian distribution, its a Ryleigh distribution after log compress. Under this situation, can we still use DDPM to learn the noise distribution?

**Comment by remidel** [comment]
Hi @Hojat-Asgariandehkordi, can you provide feedback for these questions? It would be helpful to reproduce your experiments. Thank you

**Comment by Hojat-Asgariandehkordi** [comment]
Hi @remidel and @wenbo524822927 . Thanks for the comments. I will clean the code to remove the issues and answer the questions as soon as I can.
